lib/Sema/CSApply.cpp

```diff
@@ -7527,6 +7527,39 @@ Expr *ConstraintSystem::applySolution(Solution &solution, Expr *expr,
     return nullptr;
   }
 
+  for (auto &e : solution.Conformances) {
+    const auto &conformance = e.second;
+
+    auto anchor = expr;
+    if (auto *locator = e.first)
+      anchor = locator->getAnchor();
+
+    if (conformance.isAbstract())
+      continue;
+
+    auto *concrete = conformance.getConcrete();
+    // If conformance check is not yet complete let's re-check using 'Used'
+    // to require conformance check to happen.
+    if (concrete->getState() != ProtocolConformanceState::Complete) {
+      if (auto conformance = TC.conformsToProtocol(
+              concrete->getType(), concrete->getProtocol(), DC,
+              ConformanceCheckFlags::InExpression |
+                  ConformanceCheckFlags::Used)) {
+        if (conformance->isAbstract())
+          continue;
+
+        concrete = conformance->getConcrete();
+      }
+    }
+
+    if (concrete->isComplete() && concrete->isInvalid()) {
+      TC.diagnose(anchor->getLoc(), diag::type_does_not_conform,
```

Comment:
I think it would be better to use a slightly different diagnostic here that acknowledges that the user intended to make the type conform, but it's broken, so we can't rely on it, e.g.,

  error: cannot type-check due to invalid conformance of %0 to %1

Or... should we suppress the diagnostic entirely? This thing only fails to type-check because the conformance itself was broken, so shouldn't we just be silent because we can't say anything constructive?